packages/dds/tree/src/test/feature-libraries/modular-schema/comparison.spec.ts

```diff
@@ -200,6 +201,113 @@ describe("Schema Comparison", () => {
 		);
 	});
 
+	/**
+	 * This test suite is designed to test backward compatibility when the document
```

Comment:
I think the unit tests in this suite should test more cases for different types of repo compatibility rather than be focused on types of changes which we think are backward-compatible. See e.g. how above tests use some helpers to test order and partial orders among a variety of schema. Those helpers seem like they will be easily applicable to these tests as well.

Since `allowsRepoSuperset` is mostly a simple-ish wrapper around `allowsTreeSuperset` and `allowsFieldSuperset`, I think the interesting things to verify are that it behaves in the correct direction for variations on the root field as well as variations on which sets of nodes are allowed. This is basically the point of the TODOs in the function implementation, which I think is a good goal to have this PR resolve.

Reply:
If you want to use Point/Circle or something as sample schemas to demonstrate some of these scenarios, that's fine--I don't really care. But the unit tests at this layer should also be covering cases where the schema *isn't* compatible, and for the most part the tests for `allowsFieldSuperset` or the tests for `allowsTreeSuperset` should be doing most of the heavy-lifting for compatibility cases.

Reply:
I opted not to include the Point/Circle example in this file. Perhaps it might be more appropriate to include more 'general' test cases here instead.